crates/test-framework/src/queries/validation/mod.rs

```diff
@@ -20,23 +20,49 @@ use std::{
     sync::{Arc, LazyLock},
 };
 
-use arrow::{array::RecordBatch, csv::reader::Format};
+use anyhow::{anyhow, Result};
+
+use arrow::{
+    array::{
+        Array, BooleanArray, Date32Array, Decimal128Array, Float32Array, Float64Array, Int32Array,
+        Int64Array, LargeStringArray, RecordBatch, StringArray, TimestampMicrosecondArray,
+        TimestampMillisecondArray, TimestampNanosecondArray, TimestampSecondArray, UInt16Array,
+        UInt32Array, UInt64Array, UInt8Array,
+    },
+    csv::reader::Format,
+    datatypes::TimeUnit,
+};
 use arrow::{
     csv::ReaderBuilder,
     datatypes::{DataType, SchemaRef},
 };
+use chrono::{DateTime, NaiveDate};
 
 use super::Query;
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub enum QueryValidationReason {
```

Comment:
```suggestion
pub enum QueryValidationFailReason {
```

A little clearer? I was initially confused looking at this.